### PR TITLE
cmd/swarm: do not ignore cache size 0

### LINKED ISSUE #1267: cache size=0 flag should not be ignored
currently it is not possible to disable the in memory cache of swarm since a value which equals `0` is simply ignored when starting the node:
```
	if storeCacheCapacity := ctx.GlobalUint(SwarmStoreCacheCapacity.Name); storeCacheCapacity != 0 {
		currentConfig.LocalStoreParams.CacheCapacity = storeCacheCapacity
	}
```

### PR DESCRIPTION
**Pull request by acud**
fixes https://github.com/ethersphere/go-ethereum/issues/1267

### CODE REVIEW

File: cmd/swarm/config.go [hunk 1]
```diff
@@ -259,6 +259,10 @@ func cmdLineOverride(currentConfig *bzzapi.Config, ctx *cli.Context) *bzzapi.Con
 		currentConfig.LocalStoreParams.CacheCapacity = storeCacheCapacity
 	}
 
+	if ctx.Bool(SwarmDisableCacheFlag.Name) {
```
Comment (nonsense) [inline cmd/swarm/config.go]: @justelad can't we just use the value 0 to disable it - why do we need another flag?

Comment (acud) [reply cmd/swarm/config.go]: if you omit the cache size flag - the check in the block above will yield that the value is `0`. that's why `0` value is ignored in the if statement. we can maybe change it to be a string flag and try to parse the string into an int. WDYT?

Comment (janos) [reply cmd/swarm/config.go]: Why not to use `ctx.IsSet` instead to check if the value is 0?

Comment (acud) [reply cmd/swarm/config.go]: you're right @janos. making the change now. the CLI library documentation does not explicitly mention this feature, just by looking at the examples :/

**Comment by acud** [comment]
done @janos @nonsense

**Review by nonsense — APPROVED** [review]

**Review by janos — APPROVED** [review]

**Comment by acud** [comment]
as featured on upstream: https://github.com/ethereum/go-ethereum/pull/19231

**Pull request closed without merging**